Enable Scikit-Build-Core Support

ahnaf-tahmid-chowdhury opened the pull request:
## Description
This PR introduces support for `scikit-build-core`, streamlining the build process and improving compatibility. 

> This is a minimal version of the PR, excluding manylinux support, to facilitate a faster review. The full PR can be found [here](https://github.com/openmc-dev/openmc/pull/3087).

## Key Changes
### Enable `scikit-build-core` support
- Updated `pyproject.toml` to use `scikit-build-core` instead of legacy methods.

### CMake and Build Updates
- Updated `CMakeLists.txt` to support `skbuild`, ensuring compatibility with both `scikit-build-core` and native CMake builds.
- Added `GenerateScript.cmake` to handle executable files (`openmc`) on both Linux and macOS.
- Updated `OpenMCConfig.cmake.in` to integrate `scikit-build-core`.

### Git and Packaging Improvements
- Removed ignoring of `.h5` files from `.gitignore` as they are needed.
- Removed `MANIFEST.in` as it is no longer required.

### CI and Docs Updates
- Updated Dockerfile: Removed the need to run CMake before `pip install` as `scikit-build-core` now handles it.
- Updated CI configuration files to fix build and test issues.
- Added necessary APT packages for ReadTheDocs to compile dependencies properly.

### Python API Enhancements
- Added functions to `__init__.py` to retrieve OpenMC installation configuration from the Python side.

### Sponsored By

[<img src="https://github.com/user-attachments/assets/2fe166c4-7591-447d-b63c-5e4f16e02893" width="300"  />](https://www.proximafusion.com/)
_This contribution is kindly sponsored by @proximafusion, supporting the ongoing development and modernization of OpenMC’s Python packaging and build system._

shimwell committed:
draft scikit build core wheel

shimwell committed:
renamed executable

jon-proximafusion committed:
linking libopenmc with openmc in cmakelists

jon-proximafusion committed:
Merge branch 'scikit_build_core_wheel' of github.com:shimwell/openmc into scikit_build_core_wheel

jon-proximafusion committed:
updated docs for pip install that compiles

jon-proximafusion committed:
pip install with cmake args in ci

jon-proximafusion committed:
added missing ,

jon-proximafusion committed:
no need for manual creation of build dir

shimwell committed:
[skip ci] merged develop

shimwell committed:
[skip ci] adding apt packages for read the docs

shimwell committed:
[skip ci] trying ubuntu 22.04

shimwell committed:
[skip ci] added missing s

shimwell committed:
adding gha actions

shimwell committed:
upper case ON for cmake args

shimwell committed:
added quotes to cmake args

shimwell committed:
keeping same filename

shimwell committed:
running just pytest

shimwell committed:
[skip ci] corrected cmake arg passing example

shimwell committed:
added diagnosis prints

shimwell committed:
Merge branch 'scikit_build_core_wheel' of github.com:shimwell/openmc into scikit_build_core_wheel

shimwell committed:
removed 2nd pip install which was overwritting

shimwell committed:
Merge branch 'develop' into scikit_build_core_wheel

shimwell committed:
merge in develop

jon-proximafusion committed:
matching simulation results from ci

shimwell committed:
using debug instead of on for DCMAKE_BUILD_TYPE

ahnaf-tahmid-chowdhury committed:
scikit-build-core added

ahnaf-tahmid-chowdhury committed:
Add rpath

ahnaf-tahmid-chowdhury committed:
Install by scikit-build-core

ahnaf-tahmid-chowdhury committed:
update python install doc

ahnaf-tahmid-chowdhury committed:
add support for dev mode

ahnaf-tahmid-chowdhury committed:
update dev doc with skbuild support

shimwell committed:
Merge branch 'openmc-dev:develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
Merge branch 'scikit-build-core' into scikit_build_core_wheel

jon-proximafusion committed:
back to -j4 for make

jon-proximafusion committed:
adding cmake args to pip install command

jon-proximafusion committed:
[skip ci] cmake args actual openmc options

ahnaf-tahmid-chowdhury committed:
add manylinux docker

ahnaf-tahmid-chowdhury committed:
add gcc and mpi build

ahnaf-tahmid-chowdhury committed:
Add openmc options and add more docs

ahnaf-tahmid-chowdhury committed:
install openmpi on openmpi stage

ahnaf-tahmid-chowdhury committed:
add /usr/local/lib64 to path

ahnaf-tahmid-chowdhury committed:
add workflow

ahnaf-tahmid-chowdhury committed:
remove setup.py

ahnaf-tahmid-chowdhury committed:
remove test stage

ahnaf-tahmid-chowdhury committed:
add pypi

ahnaf-tahmid-chowdhury committed:
move python deps to last

ahnaf-tahmid-chowdhury committed:
fix pytest-cov issue

ahnaf-tahmid-chowdhury committed:
fix pytest-cov issue

ahnaf-tahmid-chowdhury committed:
no need abi

ahnaf-tahmid-chowdhury committed:
fix path

ahnaf-tahmid-chowdhury committed:
move openmc to src

ahnaf-tahmid-chowdhury committed:
no need to copy lib to openmc

ahnaf-tahmid-chowdhury committed:
remove BUILD_EXAMPLES

ahnaf-tahmid-chowdhury committed:
python deps stage

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
pip install numpy

ahnaf-tahmid-chowdhury committed:
fail-fast: false

ahnaf-tahmid-chowdhury committed:
sdist.include = ["src/openmc/**"]

ahnaf-tahmid-chowdhury committed:
set DATAROOTDIR

ahnaf-tahmid-chowdhury committed:
add vendor packages

ahnaf-tahmid-chowdhury committed:
add more dirs

ahnaf-tahmid-chowdhury committed:
CMAKE_INSTALL_DATADIR

ahnaf-tahmid-chowdhury committed:
no need OpenMCTargets for skbuild

ahnaf-tahmid-chowdhury committed:
install rpath if var is defined

ahnaf-tahmid-chowdhury committed:
eval $(ncrystal-config  --setup)

ahnaf-tahmid-chowdhury committed:
sdist include *.h5

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
setup openmc lib for skbuild

ahnaf-tahmid-chowdhury committed:
prepare for merge

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
move openmc to src again

ahnaf-tahmid-chowdhury committed:
update lib paths

ahnaf-tahmid-chowdhury committed:
add include

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
add openmc.in

ahnaf-tahmid-chowdhury committed:
use OpenMC targets

ahnaf-tahmid-chowdhury committed:
install source version

ahnaf-tahmid-chowdhury committed:
add SKBUILDScripts.cmake

ahnaf-tahmid-chowdhury committed:
update get_core_path

ahnaf-tahmid-chowdhury committed:
add /usr/local/lib/ to lib path

ahnaf-tahmid-chowdhury committed:
remove file openmc_exec

ahnaf-tahmid-chowdhury committed:
remove openmc_exec

ahnaf-tahmid-chowdhury committed:
Allow CMAKE_INSTALL_RPATH_USE_LINK_PATH

ahnaf-tahmid-chowdhury committed:
renamed to GenerateScript.cmake

ahnaf-tahmid-chowdhury committed:
remove SKBUILDScripts

ahnaf-tahmid-chowdhury committed:
revert

ahnaf-tahmid-chowdhury committed:
update to 1.8

jon-proximafusion committed:
reduced diff

jon-proximafusion committed:
Merge branch 'develop' into scikit_build_core_wheel

jon-proximafusion committed:
reducing diff

ahnaf-tahmid-chowdhury committed:
prepare for merge

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
mv openmc to src

ahnaf-tahmid-chowdhury committed:
do not ignore .h5

ahnaf-tahmid-chowdhury committed:
make test

ahnaf-tahmid-chowdhury committed:
Exclude src/openmc

ahnaf-tahmid-chowdhury committed:
move openmc to python dir

ahnaf-tahmid-chowdhury committed:
python/openmc -> openmc

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
openmc -> python/openmc

ahnaf-tahmid-chowdhury committed:
add load the library from the installed module

ahnaf-tahmid-chowdhury committed:
python/openmc -> openmc

ahnaf-tahmid-chowdhury committed:
python/openmc -> openmc

ahnaf-tahmid-chowdhury committed:
add fallback

ahnaf-tahmid-chowdhury committed:
rm openmc

ahnaf-tahmid-chowdhury committed:
Add OPENMC_CORE_BASE_PATH

ahnaf-tahmid-chowdhury committed:
remove extra check

ahnaf-tahmid-chowdhury committed:
run tests on temp dir

ahnaf-tahmid-chowdhury committed:
mv openmc python/openmc

ahnaf-tahmid-chowdhury committed:
rename openmc

jon-proximafusion committed:
Merge branch 'develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
fix merge

shimwell committed:
Merge branch 'openmc-dev:develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
Update .github/workflows/pypi.yml

Co-authored-by: Jon Shimwell <jon@proximafusion.com>

ahnaf-tahmid-chowdhury committed:
Update tools/ci/manylinux.dockerfile

Co-authored-by: Jonathan Shimwell <drshimwell@gmail.com>

ahnaf-tahmid-chowdhury committed:
Update tools/ci/manylinux.dockerfile

Co-authored-by: Jonathan Shimwell <drshimwell@gmail.com>

ahnaf-tahmid-chowdhury committed:
Update docs/source/usersguide/install.rst

Co-authored-by: Jonathan Shimwell <drshimwell@gmail.com>

ahnaf-tahmid-chowdhury committed:
Update tools/ci/manylinux.dockerfile

Co-authored-by: Jonathan Shimwell <drshimwell@gmail.com>

jon-proximafusion committed:
Merge branch 'develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into scikit_build_core_wheel

ahnaf-tahmid-chowdhury committed:
add apt packages

ahnaf-tahmid-chowdhury committed:
remove manylinux docker tools

ahnaf-tahmid-chowdhury committed:
Merge branch 'openmc-dev:develop' into skbuild-minimal

shimwell committed:
Align cmake args

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

ahnaf-tahmid-chowdhury commented:
It's strange. The coverage is showing files with less coverage that are not even touched in this PR. I'm not sure how to fix this or if it's a bug.

ahnaf-tahmid-chowdhury committed:
remove comment

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

ahnaf-tahmid-chowdhury committed:
the entire second build block

ahnaf-tahmid-chowdhury committed:
remove make test

ahnaf-tahmid-chowdhury committed:
${Python_SITELIB}/openmc

ahnaf-tahmid-chowdhury committed:
move openmc paths from itit to paths.py

ahnaf-tahmid-chowdhury committed:
add test for paths

ahnaf-tahmid-chowdhury committed:
add more tests

ahnaf-tahmid-chowdhury committed:
add 100% coverage

ahnaf-tahmid-chowdhury committed:
skip warning

ahnaf-tahmid-chowdhury committed:
install editable mode

ahnaf-tahmid-chowdhury committed:
OPENMC_DEV_MODE condition

RemDelaporteMathurin commented:
Hey @ahnaf-tahmid-chowdhury thanks for this, I'm planning on testing this with @rossmacdonald98 and @SteSeg to fix our [workflow](https://github.com/LIBRA-project/BABY-LiPb/blob/main/.github/workflows/ci.yml) (see #3369 ) without having to wait for a new release or recompile from source

RemDelaporteMathurin commented:
I tried to merge `openmc-dev:develop` into `ahnaf-tahmid-chowdhury:skbuild-minimal` but ran into this issue:


python version `Python 3.13.2`

```
python -m pip install git+https://github.com/LIBRA-project/openmc@pip-with-fix
Collecting git+https://github.com/LIBRA-project/openmc@pip-with-fix
  Cloning https://github.com/LIBRA-project/openmc (to revision pip-with-fix) to /tmp/pip-req-build-ukvixuvp
  Running command git clone --filter=blob:none --quiet https://github.com/LIBRA-project/openmc /tmp/pip-req-build-ukvixuvp
  Running command git checkout -b pip-with-fix --track origin/pip-with-fix
  Switched to a new branch 'pip-with-fix'
  branch 'pip-with-fix' set up to track 'origin/pip-with-fix'.
  Resolved https://github.com/LIBRA-project/openmc to commit 4c4ac3408b3eb8e762751d6783dd00c529046b11
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy (from openmc==0.0.0)
  Downloading numpy-2.2.5-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (62 kB)
Collecting h5py (from openmc==0.0.0)
  Downloading h5py-3.13.0-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.5 kB)
Collecting scipy (from openmc==0.0.0)
  Using cached scipy-1.15.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)
Collecting ipython (from openmc==0.0.0)
  Downloading ipython-9.1.0-py3-none-any.whl.metadata (4.4 kB)
Collecting matplotlib (from openmc==0.0.0)
  Using cached matplotlib-3.10.1-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (11 kB)
Collecting pandas (from openmc==0.0.0)
  Using cached pandas-2.2.3-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (89 kB)
Collecting lxml (from openmc==0.0.0)
  Downloading lxml-5.4.0-cp313-cp313-manylinux_2_28_x86_64.whl.metadata (3.5 kB)
Collecting uncertainties (from openmc==0.0.0)
  Downloading uncertainties-3.2.3-py3-none-any.whl.metadata (7.0 kB)
Requirement already satisfied: setuptools in /home/remidm/anaconda3/envs/temp-openmc-build/lib/python3.13/site-packages (from openmc==0.0.0) (75.8.0)
Collecting endf (from openmc==0.0.0)
  Downloading endf-0.1.4.tar.gz (80 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting decorator (from ipython->openmc==0.0.0)
  Using cached decorator-5.2.1-py3-none-any.whl.metadata (3.9 kB)
Collecting ipython-pygments-lexers (from ipython->openmc==0.0.0)
  Using cached ipython_pygments_lexers-1.1.1-py3-none-any.whl.metadata (1.1 kB)
Collecting jedi>=0.16 (from ipython->openmc==0.0.0)
  Using cached jedi-0.19.2-py2.py3-none-any.whl.metadata (22 kB)
Collecting matplotlib-inline (from ipython->openmc==0.0.0)
  Using cached matplotlib_inline-0.1.7-py3-none-any.whl.metadata (3.9 kB)
Collecting pexpect>4.3 (from ipython->openmc==0.0.0)
  Using cached pexpect-4.9.0-py2.py3-none-any.whl.metadata (2.5 kB)
Collecting prompt_toolkit<3.1.0,>=3.0.41 (from ipython->openmc==0.0.0)
  Downloading prompt_toolkit-3.0.51-py3-none-any.whl.metadata (6.4 kB)
Collecting pygments>=2.4.0 (from ipython->openmc==0.0.0)
  Using cached pygments-2.19.1-py3-none-any.whl.metadata (2.5 kB)
Collecting stack_data (from ipython->openmc==0.0.0)
  Using cached stack_data-0.6.3-py3-none-any.whl.metadata (18 kB)
Collecting traitlets>=5.13.0 (from ipython->openmc==0.0.0)
  Using cached traitlets-5.14.3-py3-none-any.whl.metadata (10 kB)
Collecting contourpy>=1.0.1 (from matplotlib->openmc==0.0.0)
  Downloading contourpy-1.3.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.5 kB)
Collecting cycler>=0.10 (from matplotlib->openmc==0.0.0)
  Using cached cycler-0.12.1-py3-none-any.whl.metadata (3.8 kB)
Collecting fonttools>=4.22.0 (from matplotlib->openmc==0.0.0)
  Using cached fonttools-4.57.0-cp313-cp313-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (102 kB)
Collecting kiwisolver>=1.3.1 (from matplotlib->openmc==0.0.0)
  Using cached kiwisolver-1.4.8-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.2 kB)
Collecting packaging>=20.0 (from matplotlib->openmc==0.0.0)
  Using cached packaging-25.0-py3-none-any.whl.metadata (3.3 kB)
Collecting pillow>=8 (from matplotlib->openmc==0.0.0)
  Using cached pillow-11.2.1-cp313-cp313-manylinux_2_28_x86_64.whl.metadata (8.9 kB)
Collecting pyparsing>=2.3.1 (from matplotlib->openmc==0.0.0)
  Using cached pyparsing-3.2.3-py3-none-any.whl.metadata (5.0 kB)
Collecting python-dateutil>=2.7 (from matplotlib->openmc==0.0.0)
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting pytz>=2020.1 (from pandas->openmc==0.0.0)
  Downloading pytz-2025.2-py2.py3-none-any.whl.metadata (22 kB)
Collecting tzdata>=2022.7 (from pandas->openmc==0.0.0)
  Downloading tzdata-2025.2-py2.py3-none-any.whl.metadata (1.4 kB)
Collecting parso<0.9.0,>=0.8.4 (from jedi>=0.16->ipython->openmc==0.0.0)
  Using cached parso-0.8.4-py2.py3-none-any.whl.metadata (7.7 kB)
Collecting ptyprocess>=0.5 (from pexpect>4.3->ipython->openmc==0.0.0)
  Using cached ptyprocess-0.7.0-py2.py3-none-any.whl.metadata (1.3 kB)
Collecting wcwidth (from prompt_toolkit<3.1.0,>=3.0.41->ipython->openmc==0.0.0)
  Using cached wcwidth-0.2.13-py2.py3-none-any.whl.metadata (14 kB)
Collecting six>=1.5 (from python-dateutil>=2.7->matplotlib->openmc==0.0.0)
  Using cached six-1.17.0-py2.py3-none-any.whl.metadata (1.7 kB)
Collecting executing>=1.2.0 (from stack_data->ipython->openmc==0.0.0)
  Using cached executing-2.2.0-py2.py3-none-any.whl.metadata (8.9 kB)
Collecting asttokens>=2.1.0 (from stack_data->ipython->openmc==0.0.0)
  Using cached asttokens-3.0.0-py3-none-any.whl.metadata (4.7 kB)
Collecting pure-eval (from stack_data->ipython->openmc==0.0.0)
  Using cached pure_eval-0.2.3-py3-none-any.whl.metadata (6.3 kB)
Downloading h5py-3.13.0-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.9/4.9 MB 32.1 MB/s eta 0:00:00
Downloading numpy-2.2.5-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.1/16.1 MB 37.0 MB/s eta 0:00:00
Downloading ipython-9.1.0-py3-none-any.whl (604 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 604.1/604.1 kB 22.2 MB/s eta 0:00:00
Downloading lxml-5.4.0-cp313-cp313-manylinux_2_28_x86_64.whl (4.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.9/4.9 MB 38.7 MB/s eta 0:00:00
Using cached matplotlib-3.10.1-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (8.6 MB)
Using cached pandas-2.2.3-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.7 MB)
Using cached scipy-1.15.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (37.3 MB)
Downloading uncertainties-3.2.3-py3-none-any.whl (60 kB)
Downloading contourpy-1.3.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (322 kB)
Using cached cycler-0.12.1-py3-none-any.whl (8.3 kB)
Using cached fonttools-4.57.0-cp313-cp313-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.8 MB)
Using cached jedi-0.19.2-py2.py3-none-any.whl (1.6 MB)
Using cached kiwisolver-1.4.8-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.5 MB)
Using cached packaging-25.0-py3-none-any.whl (66 kB)
Using cached pexpect-4.9.0-py2.py3-none-any.whl (63 kB)
Using cached pillow-11.2.1-cp313-cp313-manylinux_2_28_x86_64.whl (4.6 MB)
Downloading prompt_toolkit-3.0.51-py3-none-any.whl (387 kB)
Using cached pygments-2.19.1-py3-none-any.whl (1.2 MB)
Using cached pyparsing-3.2.3-py3-none-any.whl (111 kB)
Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Downloading pytz-2025.2-py2.py3-none-any.whl (509 kB)
Using cached traitlets-5.14.3-py3-none-any.whl (85 kB)
Downloading tzdata-2025.2-py2.py3-none-any.whl (347 kB)
Using cached decorator-5.2.1-py3-none-any.whl (9.2 kB)
Using cached ipython_pygments_lexers-1.1.1-py3-none-any.whl (8.1 kB)
Using cached matplotlib_inline-0.1.7-py3-none-any.whl (9.9 kB)
Using cached stack_data-0.6.3-py3-none-any.whl (24 kB)
Using cached asttokens-3.0.0-py3-none-any.whl (26 kB)
Using cached executing-2.2.0-py2.py3-none-any.whl (26 kB)
Using cached parso-0.8.4-py2.py3-none-any.whl (103 kB)
Using cached ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
Using cached six-1.17.0-py2.py3-none-any.whl (11 kB)
Using cached pure_eval-0.2.3-py3-none-any.whl (11 kB)
Using cached wcwidth-0.2.13-py2.py3-none-any.whl (34 kB)
Building wheels for collected packages: openmc, endf
  Building wheel for openmc (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for openmc (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [63 lines of output]
      2025-04-23 13:03:43,589 - scikit_build_core - INFO - RUN: /tmp/pip-build-env-j5ei7hhk/normal/lib/python3.13/site-packages/cmake/data/bin/cmake -E capabilities
      2025-04-23 13:03:43,591 - scikit_build_core - INFO - CMake version: 4.0.0
      *** scikit-build-core 0.11.1 using CMake 4.0.0 (wheel)
      2025-04-23 13:03:43,593 - scikit_build_core - INFO - Build directory: /tmp/tmpzh05vedz/build
      *** Configuring CMake...
      2025-04-23 13:03:43,599 - scikit_build_core - INFO - RUN: ninja --version
      2025-04-23 13:03:43,600 - scikit_build_core - INFO - Ninja version: 1.11.1
      2025-04-23 13:03:43,601 - scikit_build_core - INFO - RUN: /tmp/pip-build-env-j5ei7hhk/normal/lib/python3.13/site-packages/cmake/data/bin/cmake -S. -B/tmp/tmpzh05vedz/build -DCMAKE_BUILD_TYPE:STRING=Release -C/tmp/tmpzh05vedz/build/CMakeInit.txt -DCMAKE_INSTALL_PREFIX=/tmp/tmpzh05vedz/wheel/platlib/openmc -DCMAKE_MAKE_PROGRAM=ninja
      loading initial cache file /tmp/tmpzh05vedz/build/CMakeInit.txt
      -- The C compiler identification is GNU 13.3.0
      -- The CXX compiler identification is GNU 13.3.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/gcc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/g++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Found Git: /usr/bin/git (found version "2.43.0")
      -- Using git describe for versioning
      CMake Warning at cmake/Modules/GetVersionFromGit.cmake:46 (message):
        No git tags found.  Version set to 0.0.0.
      
        Run 'git fetch --tags' to ensure proper versioning.
      
        For more information, see OpenMC developer documentation.
      Call Stack (most recent call first):
        CMakeLists.txt:7 (include)
      
      
      -- OpenMC version: 0.0.0
      -- OpenMC dev state: false
      -- OpenMC commit hash: 4c4ac3408b3eb8e762751d6783dd00c529046b11
      -- OpenMC commit count:
      -- OPENMC_USE_OPENMP ON
      -- OPENMC_BUILD_TESTS ON
      -- OPENMC_ENABLE_PROFILE OFF
      -- OPENMC_ENABLE_COVERAGE OFF
      -- OPENMC_USE_DAGMC OFF
      -- OPENMC_USE_LIBMESH OFF
      -- OPENMC_USE_MPI OFF
      -- OPENMC_USE_MCPL OFF
      -- OPENMC_USE_UWUW OFF
      -- Found OpenMP_C: -fopenmp (found version "4.5")
      -- Found OpenMP_CXX: -fopenmp (found version "4.5")
      -- Found OpenMP: TRUE (found version "4.5")
      -- Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
      -- Could NOT find PNG (missing: PNG_LIBRARY PNG_PNG_INCLUDE_DIR)
      CMake Error at /tmp/pip-build-env-j5ei7hhk/normal/lib/python3.13/site-packages/cmake/data/share/cmake-4.0/Modules/FindPackageHandleStandardArgs.cmake:227 (message):
        Could NOT find HDF5 (missing: HDF5_LIBRARIES HDF5_INCLUDE_DIRS
        HDF5_HL_LIBRARIES C HL)
      Call Stack (most recent call first):
        /tmp/pip-build-env-j5ei7hhk/normal/lib/python3.13/site-packages/cmake/data/share/cmake-4.0/Modules/FindPackageHandleStandardArgs.cmake:591 (_FPHSA_FAILURE_MESSAGE)
        /tmp/pip-build-env-j5ei7hhk/normal/lib/python3.13/site-packages/cmake/data/share/cmake-4.0/Modules/FindHDF5.cmake:1024 (find_package_handle_standard_args)
        CMakeLists.txt:173 (find_package)
      
      
      -- Configuring incomplete, errors occurred!
      
      *** CMake configuration failed
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for openmc
  Building wheel for endf (pyproject.toml) ... done
  Created wheel for endf: filename=endf-0.1.4-cp313-cp313-linux_x86_64.whl size=114793 sha256=b6a015388bae35b446b1bf35dafced1ed062b20156ed93079796c7fbbdcc2aba
  Stored in directory: /home/remidm/.cache/pip/wheels/39/cf/5b/92b26adc097196ccabee43fa54b2fd84d088001e7ac2354671
Successfully built endf
Failed to build openmc
ERROR: Failed to build installable wheels for some pyproject.toml based projects (openmc)
```

ahnaf-tahmid-chowdhury commented:
Hello, @RemDelaporteMathurin. This is a short PR that only introduces the scikit-build-core setup. You may need to install dependencies before running the setup.

RemDelaporteMathurin commented:
> Yes in the future (follow up PR) these will be bundled in the wheel but for now if you install via pip and this branch I guess these are needed.
> 
> ```
> sudo apt install g++ cmake libhdf5-dev libpng-dev
> ```

Installed these dependencies and I still have the error. Is there a list somewhere?

ahnaf-tahmid-chowdhury commented:
For minimal setup you may only need hdf5 installed. For detail setup you may consider reviewing the CI workflows. There is a Manylinux dockerfile available at #3087 that might also help.

ahnaf-tahmid-chowdhury committed:
Merge branch 'openmc-dev:develop' into skbuild-minimal

ahnaf-tahmid-chowdhury committed:
Merge branch 'openmc-dev:develop' into skbuild-minimal

ahnaf-tahmid-chowdhury commented:
We need to do the same thing for mcpl that we have done for ncrystal.

ahnaf-tahmid-chowdhury commented:
I have already created PR #3429 to make this PR working again.

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

ahnaf-tahmid-chowdhury commented:
Everything is working now.

ahnaf-tahmid-chowdhury committed:
mcpl is optional now

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

ahnaf-tahmid-chowdhury committed:
use dagmc suggestion as example

ahnaf-tahmid-chowdhury commented on docs/source/usersguide/install.rst:
```diff
@@ -488,29 +490,72 @@ To run the test suite, you will first need to download a pre-generated cross
-    python -m pip install .
+.. code-block:: bash
+
+    python -m pip install . --config-settings=cmake.args="-DOPENMC_USE_MPI=ON;-DOPENMC_USE_MCPL=ON"
```
Nice catch

ahnaf-tahmid-chowdhury committed:
no need to set LD_LIBRARY_PATH

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

ahnaf-tahmid-chowdhury committed:
fix find_package

shimwell commented:
Ah looks like the recent changes to the coverage need merging into this PR

CI error is 
Run # Merge C++ and Python LCOV into a single file for upload
cat: coverage-cpp.lcov: No such file or directory

I think this is the PR that got the CI working again https://github.com/openmc-dev/openmc/pull/3594

ahnaf-tahmid-chowdhury committed:
build-dir="build"

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

ahnaf-tahmid-chowdhury commented:
This is just the foundation. More features will be added later, such as using CMake’s built-in modules for target generation, adding an `openmc-config` CLI, supporting manylinux images for PyPI, and improving the handling of external packages.

ahnaf-tahmid-chowdhury committed:
Merge remote-tracking branch 'upstream/develop' into skbuild-minimal

GuySten reviewed (approved):
This looks good to me.

paulromano reviewed (changes requested):
Putting a placeholder as I'd like to review this before we merge

GuySten reviewed (changes requested):
I noticed that the cpp test suite does not run.

shimwell committed:
Merge remote-tracking branch 'upstream/develop' into sync-skbuild-minimal

# Conflicts:
#	cmake/OpenMCConfig.cmake.in
#	pyproject.toml
#	tools/ci/gha-install.py

shimwell commented:
I have just updated this branch as there were some conflicts that had crept in. updated with latest develop

shimwell committed:
Tell scikit-build-core where the version comes from

Every job in the CI matrix fails at about two minutes with

    AssertionError: project.version is not specified, must be statically
    present or tool.scikit-build metadata.version.provider configured
    when dynamic

project.version is declared dynamic and [tool.setuptools_scm] is
present, but under the setuptools backend that was enough on its own,
whereas scikit-build-core requires an explicit provider. Without one it
refuses to generate metadata at all, so the failure happens before any
compilation, which is why every matrix entry fails in the same couple of
minutes and why readthedocs fails too.

Uses the top-level [[tool.dynamic-metadata]] spelling. The older
tool.scikit-build.metadata.version form also works but scikit-build-core
1.x warns that it is deprecated. Both were checked; only the new one is
warning free. The top-level form needs scikit-build-core 0.10 or newer,
so requires is pinned accordingly, otherwise an older release would
ignore the table and fail with the same assertion.

Verified locally: metadata now resolves to
openmc-0.15.4.dev386+gf82a19308 with Requires-Python >=3.12 and
License-Expression MIT.

shimwell commented:
fixed version

GuySten commented:
I think this PR still does not run the cpp test suite.

shimwell committed:
ci: run the C++ unit tests again under scikit-build-core

The scikit-build-core switch dropped the 'make test' invocation from the
test step. The C++ unit tests were still compiled but never executed.

scikit-build-core drives CMake with the Ninja generator, so 'make test'
no longer works. It configures the build in $GITHUB_WORKSPACE/build,
the same directory the gcovr coverage step already reads, so ctest can
be pointed straight at it. --no-tests=error is used because ctest
otherwise exits 0 when it finds no tests at all.

shimwell commented:
> I think this PR still does not run the cpp test suite.

I have now tried to add these cpp tests

ahnaf-tahmid-chowdhury commented:
Maybe we should keep this minimal so it’s easier to review. It’s been a while, and we’re still here.
